src/core/Button/style.ts

```diff
@@ -8,16 +8,44 @@ import {
   Props,
 } from "../styles";
 
-const sdsPropNames = ["isAllCaps", "isRounded", "sdsStyle", "sdsType"];
+export interface ExtraProps {
+  isAllCaps?: boolean;
+  isRounded?: boolean;
+  sdsStyle?: "minimal" | "rounded" | "square";
+  sdsType?: "primary" | "secondary";
+  /**
+   * TODO(thuang): Remove custom `color` prop when we upgrade to MUIv5.
+   * Currently we're extending MUIv4 Button's `color` props from "primary" and
+   * "secondary" to the following ones.
+   */
+  color?: "primary" | "secondary" | "success" | "error" | "warning" | "info";
+}
+
+type V5ButtonProps = Props & ExtraProps & Omit<ButtonProps, "color">;
+
+// Please keep this in sync with the props used in `ExtraProps`
+const doNotForwardProps = [
```

Comment:
perhaps we could rename this to `nonMuiProps` or something along those lines?  That is a little more expressive around why we don't want to forward these props.  (I'm happy to make this change but wanted to run it by you first).